include/circt/Dialect/RTL/Structure.td

```diff
@@ -0,0 +1,88 @@
+def RTLModuleOp : RTLOp<"module",
+      [IsolatedFromAbove, FunctionLike, Symbol, RegionKindInterface,
+       SingleBlockImplicitTerminator<"DoneOp">]> {
+  let summary = "RTL Module";
+  let description = [{
+    The "rtl.module" operation represents a Verilog module, including a given
+    name, a list of ports, and a body that represents the connections within
+    the module.
+  }];
+  let arguments = (ins);
+  let results = (outs);
+  let regions = (region SizedRegion<1>:$body);
+
+  let skipDefaultBuilders = 1;
+  let builders = [
+    OpBuilder<"OpBuilder &builder, OperationState &result, StringAttr name, "
+              "ArrayRef<std::tuple<StringAttr, Type, StringAttr>> ports">
+  ];
+
+  let extraClassDeclaration = [{
+    using FunctionLike::getBody;
+
+    // Implement RegionKindInterface.
+    static RegionKind getRegionKind(unsigned index) { return RegionKind::Graph; }
+
+    void getRTLModulePortInfo(Operation *op, SmallVectorImpl<RTLModulePortInfo> &results);
+    
+    // // Decode information about the input and output ports on this module.
+    void getRTLPortInfo(SmallVectorImpl<RTLModulePortInfo> &results) {
+      getRTLModulePortInfo(*this, results);
+    }
+
+    // TODO(mlir): FunctionLike shouldn't produce a getBody() helper, it is
```

Comment:
Is this TODO specific to this file?

Reply:
This came from the FIRRTL module - we should change the code generated by the FunctionLike trait to use a different name for the helper (in my opinion).  I'm curious to know what @River707 thinks about this, and how we deal with name conflicts in general from traits.